subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -478,32 +485,49 @@ static int __wifi_args_to_params(size_t argc, char *argv[],
 			return -EINVAL;
 		}
 
-		if (iface_mode == WIFI_MODE_INFRA && channel == 0) {
-			params->channel = WIFI_CHANNEL_ANY;
+		if (iface_mode == WIFI_MODE_INFRA) {
+			if (channel < 0) {
+				/* Negative channel means band */
+				switch (-channel) {
+				case 2:
+					params->band = WIFI_FREQ_BAND_2_4_GHZ;
+					break;
+				case 5:
+					params->band = WIFI_FREQ_BAND_5_GHZ;
+					break;
+				case 6:
+					params->band = WIFI_FREQ_BAND_6_GHZ;
+					break;
+				default:
+					print(context.sh, SHELL_ERROR,
+					      "Invalid band: %ld\n", channel);
+					return -EINVAL;
+				}
+			}
 		} else {
-			const uint8_t bands[] = {WIFI_FREQ_BAND_2_4_GHZ,
-					       WIFI_FREQ_BAND_5_GHZ,
-					       WIFI_FREQ_BAND_6_GHZ};
-			uint8_t band;
-			bool found = false;
-			char bands_str[MAX_BANDS_STR_LEN] = {0};
-			size_t offset = 0;
-
-			for (band = 0; band < ARRAY_SIZE(bands); band++) {
+			if (channel < 0) {
+				print(context.sh, SHELL_ERROR,
+				      "Invalid channel: %ld\n", channel);
+				return -EINVAL;
+			}
+		}
+
+		if (channel > 0) {
+			for (band = 0; band < ARRAY_SIZE(all_bands); band++) {
 				offset += snprintf(bands_str + offset,
-						  sizeof(bands_str) - offset,
-						  "%s%s",
-						  band ? "," : "",
-						  wifi_band_txt(bands[band]));
+							sizeof(bands_str) - offset,
+							"%s%s",
+							band ? "," : "",
+							wifi_band_txt(all_bands[band]));
 				if (offset >= sizeof(bands_str)) {
 					print(context.sh, SHELL_ERROR,
-					      "Failed to parse channel: %s: "
-					      "band string too long\n",
-					      argv[idx]);
+						"Failed to parse channel: %s: "
+						"band string too long\n",
+						argv[idx]);
```

Comment:
Are these changes necessary? If so, maybe they should be part of a separate commit?